duckduckgo/chrome-zeroclickinfo
# subdomain 1st party

**Reviewer:**



## Description:


Two changes here:

1. When looking for first party requests we need to consider subdomains. For example `gs.statcounter.com` and `statcounter.com`. I added an isFirstPartyRequest function to handle this.

2. Fixed a bug in looking up related entities in `isRelatedEntity` that was also not considering subdomains. I'm joining the entity properties list in to a regex and using that to match against the request hostname. 

## Steps to test this PR:


To test `isFirstPartyRequest`
Go to statcounter.com. The page should correctly load and not be blocking any statcounter.com trackers

To test `isRelatedEntity`
Go to adwords.google.com. The popup should not show any google trackers being blocked

## Automated tests:

###### Reviewer Checklist:


###### PR Author Checklist:
